public/js/javascript.js

```diff
@@ -1,24 +1,141 @@
+            });
+
+            /**
+             * AJAX image upload
+             */
+            $("#imgajax").change(function (e) {
+                var $form = $("#content").submit();
+                $.ajax({
+                    url: $form.attr("action"),
+                    type: "POST",
+                    data: new FormData($form[0]),
+                    processData:false,
+                    contentType:  false,
+                    cache: false,
+                }).done(function (result, request, headers) {
+                    var $resp = $.parseJSON(result);
+                    ajaxImageUpload.getAjaxResponse($resp["successFiles"], "p", ".demo-box", "image-upload-message successFiles");
+                    ajaxImageUpload.getAjaxResponse($resp["errorFiles"], "p", ".demo-box", "image-upload-message errorFiles");
+                    ajaxImageUpload.showFiles();
+                }).fail(function (result, request, headers) {
+                    console.error("Error:", result.responseText);
+                });
+                // clear file input
+                $("#imgajax").replaceWith($("#imgajax").val('').clone(true));
+                e.preventDefault();
+                return false;
+            });
+        },
+
+        /**
+         * Create DOM nodes with text, class and appends them to elementAppend
+         */
+        showMessages: function (text, elementCreate, elementAppend, className) {
+            var el = document.createElement(elementCreate);
+            el.className += className;
+            el.innerHTML = text;
+
+            $(elementAppend).append(el).slideDown(1000, function (e) {
+                e.preventDefault();
+                setTimeout(function() {
+                    $(elementCreate).slideUp(1000, function () {
+                        $(this).fadeOut("slow", function () {
+                            $(this).remove();
+                         });
+                    });
+                }, 6000);
+            });
+        },
+
+        /**
+         * Show AJAX reponse
+         */
+        getAjaxResponse: function (response, elementCreate, elementAppend, className) {
+            if (typeof response !== "undefined" && typeof response !== undefined) {
+                $(elementAppend).append($("<div class='dinamicly-div-append-wrapper'></div>"));
+                $.each(response, function(key, text) {
+                    ajaxImageUpload.showMessages(text, elementCreate, 'div.dinamicly-div-append-wrapper', className);
+                });
+            }
+        },
+
+        /**
+         * Gallery view
+         */
+        showFiles: function () {
+                $(".uploader-inline, .large-image").hide();
+                $(".gallery-view").find("figure.centered").not(".large-image").remove();
+                $(".gallery-view, .ajax-loader").show();
+            $.get( "/learn-zf2-ajax-image-gallery/index/files", function (files) {

```

Comment:
we should avoid using "/" directly, you can add : 
```php
<script type="text/javascript">
var basePath = '<?php echo $this->basePath(); ?>';
</script>
```
in <code>Application/view/layout/header.phtml</code> before <code>headScript</code> view helper called, and after that, you can use : 
```javascript
$.get(basePath +  "/learn-zf2-ajax-image-gallery/index/files", function (files) {
```